packages/@sanity/preview/src/components/RenderPreviewSnapshot.tsx

```diff
@@ -28,13 +28,14 @@ export default function RenderPreviewSnapshot(props: Props) {
 
   // TODO: Bjoerge: Check for image type with "is()"
   const renderAsBlockImage = layout === 'block' && type && type.name === 'image'
-
+  const typeName = snapshot?._type
+  const icon = (type.to && (type.to || []).find(t => t.name === typeName)?.icon) || (type && type.icon)
```

Comment:
This could be simplified a bit
```suggestion
  const icon = (type.to && type.to.find(t => t.name === typeName)?.icon) || type.icon
```